collect/cache/cuckooSentCache.go

```diff
@@ -28,18 +28,29 @@ type keptTraceCacheEntry struct {
 	reason         uint32 // which rule was used to decide to keep the trace
 }
 
-func NewKeptTraceCacheEntry(trace *types.Trace) *keptTraceCacheEntry {
-	if trace == nil {
+type TraceInfo interface {
+	ID() string
+	SampleRate() uint
+	DescendantCount() uint32
+	SpanEventCount() uint32
+	SpanLinkCount() uint32
+	SpanCount() uint32
+	SetSentReason(uint)
```

Comment:
Since we are including the sentReasonCache inside the sent trace cache, we need a way to update the sent reason on the trace object inside the cache